### PR TITLE
Speedups for GitHub Actions

### PR DESCRIPTION
**Pull request by mehrdadn**
## Why are these changes needed?

- Uses built-in Python installations to build wheels instead of Anaconda, to avoid slowing down builds on Windows.

- Skips configuring AWS credentials on forked repos. (This speeds up builds because the configuration on macOS takes a long time to eventually produce an error.)

- Adds GitHub Actions caching of pip packages. This helps reduce network traffic to external servers (which reduces chances of CI failure) and also helps with recycling some non-binary wheels that pip builds locally (which can be a slow process).

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -60,11 +63,32 @@ jobs:
...
       uses: aws-actions/configure-aws-credentials@v1
       with:
         aws-access-key-id: ${{ secrets.GHA_AWS_ACCESS_KEY_ID }}
         aws-secret-access-key: ${{ secrets.GHA_AWS_SECRET_ACCESS_KEY }}
         aws-region: us-west-2
+    - name: Prepare caching
+      id: info
+      shell: bash
+      run: |
+        case "${OSTYPE}" in linux*) pip_cache=~/.cache/pip;; darwin*) pip_cache=~/Library/Caches/pip;; msys) pip_cache="${LOCALAPPDATA}\pip\Cache";; esac
```
Comment (simon-mo) [inline .github/workflows/main.yml:76]: can you make this multiline for more readability?

File: python/build-wheel-windows.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      cd "${WORKSPACE_DIR}"/python
+      python setup.py --quiet bdist_wheel
+      pip uninstall -y ray
+    )
+  done
+
+  "${WORKSPACE_DIR}"/ci/travis/bazel-preclean.sh
+  if [ 0 -eq "${ray_uninstall_status}" ]; then  # If Ray was previously installed, restore it
+    install_ray
+  fi
```
Comment (simon-mo) [inline python/build-wheel-windows.sh:72]: User can install un their own. Do we need this in our CI?

**Review by simon-mo — APPROVED** [review]
LGTM

File: python/build-wheel-windows.sh (hunk 2 above)
Comment (mehrdadn) [reply python/build-wheel-windows.sh:72]: This isn't for users, it's or our CI. Whether we build the wheels and then do testing is in an arbitrary order (currently we build wheels first, then test, but we can easily swap it around later) so this is needed to ensure any installation remains after wheels are built.

File: .github/workflows/main.yml (hunk 1 above)
Comment (mehrdadn) [reply .github/workflows/main.yml:76]: Sure

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -60,11 +63,36 @@ jobs:
         fetch-depth: 0
     - name: Configure AWS Credentials
       continue-on-error: true
+      if: github.repository == 'ray-project/ray'
```
Comment (simon-mo) [inline .github/workflows/main.yml:66]: also not a pull request?

File: ci/travis/ci.sh [hunk 4]
```diff
@@ -452,6 +423,7 @@ init() {
 
 build() {
   if ! need_wheels; then
+    bazel_ensure_buildable
```
Comment (simon-mo) [inline ci/travis/ci.sh:426]: this is ran both in need_wheels and !need_wheels, can we change it to `bazel_ensure_buidable_on_windows` or something and run it out side of the if condition

File: .github/workflows/main.yml (hunk 3 above)
Comment (mehrdadn) [reply .github/workflows/main.yml]: Done

File: ci/travis/ci.sh (hunk 4 above)
Comment (mehrdadn) [reply ci/travis/ci.sh]: Done

**Pull request merged**